Makefile

```diff
@@ -80,12 +107,22 @@ ELF2ROM    := tools/elf2rom
 ZAPD       := tools/ZAPD/ZAPD.out
 FADO       := tools/fado/fado.elf
 
-OPTFLAGS := -O2
+ifeq ($(COMPILER),gcc)
+  OPTFLAGS := -Os -ffast-math -fno-unsafe-math-optimizations
+else
+  OPTFLAGS := -O2
+endif
+
 ASFLAGS := -march=vr4300 -32 -Iinclude
-MIPS_VERSION := -mips2
 
-# we support Microsoft extensions such as anonymous structs, which the compiler does support but warns for their usage. Surpress the warnings with -woff.
-CFLAGS += -G 0 -non_shared -Xfullwarn -Xcpluscomm $(INC) -Wab,-r4300_mul -woff 649,838,712
+ifeq ($(COMPILER),gcc)
+  CFLAGS += -G 0 -nostdinc $(INC) -DAVOID_UB -march=vr4300 -mfix4300 -mabi=32 -mno-abicalls -mdivide-breaks -fno-zero-initialized-in-bss -fno-toplevel-reorder -ffreestanding -fno-common -fno-merge-constants -mno-explicit-relocs -mno-split-addresses $(CHECK_WARNINGS) -funsigned-char
+  MIPS_VERSION := -mips3
+else 
```

Comment:
Since the linker is explicitly adding debugging sections, shouldn't we add debugging flags when compiling with gcc?

Reply:
feels like something to me the user should opt into (by adding it themsleves) if they want to make use of it.
The changes in the linker script simply say to include them if they are there, doesnt mean they have to be (imo)

Reply:
According to the gcc manual, the default debugging level is 2. https://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html#Debugging-Options

But I don't really trust gcc documentation, so I did a few manual tests.

(Using `mips-linux-gnu-gcc` + `-S` (produce assembly instead of a .o file) + the flags described here )

no `-gN` behaves the same as `-g0`. `-g1` adds a bit of ~~bloat~~ debugging information. `-g2` goes nuts and `-g3` is a bit absurd...

Conclusion: the documentation is lying again.

Reply:
> feels like something to me the user should opt into (by adding it themsleves) if they want to make use of it. The changes in the linker script simply say to include them if they are there, doesnt mean they have to be (imo)

Yeah, no worries. I just though the flag was intended to be used, but somebody forgot to add it.